Automattic/jetpack
# Build: Simplify Dashboard Build Toolchain

WIP. Prep work for #12072.

One core thing I've identified is that `tools/builder/react.js` uses `static.jsx` to produce minimal static markup (HTML) versions of the dashboard (really just "Turn on your JavaScript" notices). It `require()`s the webpack-bundled version of `static.jsx` to that end, and attaches `renderToStaticMarkup()`-produced strings to `window`, which it then writes to HTML files.

Getting rid of the complexity added by passing information by attaching to `window` (and thus requiring `jsdom` etc) is the main objective of this PR.

Maybe we can instead produce those HTML files as by-products of the bundling, i.e. through a Webpack plugin or loader? https://github.com/GoogleChromeLabs/prerender-loader#string-prerendering looks very promising!

Alternatively, we shouldn't bundle but just transpile `static.jsx` so we can simply `require()` it from `tools/builder/react.js`.

#### Changes proposed in this Pull Request:

Don't use `window`/`jsdom` in build system.

#### Testing instructions:

```
yarn distclean && yarn
yarn build-client
yarn docker:up
```

- Navigate to `/wp-admin/admin.php?page=jetpack_modules`
- Disable JavaScript
- See this message?

![image](https://user-images.githubusercontent.com/96308/56582883-19c3c300-65fa-11e9-92d8-e3355f73317d.png)

#### Proposed changelog entry for your changes:

Build: Simplify Dashboard Build Toolchain
